tasks/main.yml

```diff
@@ -0,0 +1,25 @@
+---
+
+- name: Find required providers
+  set_fact:
+    providers: |
+      {{
+        certificate_requests |
+        map(attribute='provider', default=__certificate_provider_default) |
+        unique |
+        list
+      }}
+
+- name: Ensure certmonger is installed
+  package:
+    name: certmonger
+    state: present
+  when: "'certmonger' in providers"
```

Comment:
This hasn't been merged yet, but we have best practices for provider-specific tasks and variables: https://github.com/oasis-roles/meta_standards/blob/7fd94fcbb9388b2f28d30b59b1c85d805a9ed062/README.md#supporting-multiple-providers

I suppose you could have two tasks for this - something like this:
```yaml
- name: Load provider specific variables
  # these files are in the role vars/ directory
  include_vars: "{{ __certificate_provider ~ '.yml' }}"
  loop: "{{ __certificate_providers }}"
  loop_control:
    loop_var: __certificate_provider

- name: Perform provider specfic tasks
  # these files are in the role tasks/ directory
  include_tasks: "main-{{ __certificate_provider ~ '.yml' }}"
  loop: "{{ __certificate_providers }}"
  loop_control:
    loop_var: __certificate_provider
```

In `vars/certmonger.yml` you would have
```yaml
__certificate_certmonger_packages:
  - certmonger
__certificate_certmonger_services:
  - certmonger
```

In `tasks/main-certmonger.yml` you would have
```yaml
- name: Ensure certmonger packages are installed
  package:
    name: "{{ __certificate_certmonger_packages }}"
    state: present

- name: Ensure certmonger services are enabled and started
  service:
    name: "{{ item }}"
    state: started
    enabled: yes
  loop: "{{ __certificate_certmonger_services }}"
```

The handler is going to be tricky, since you may need to invoke multiple handlers - something like
```yaml
- name: certificate handler for certmonger
  do something for certmonger
  listen: __certificate_request_handler
  when: 'certmonger' in __certificate_providers

- name: certificate handler for xxx
  do something for xxx provider
  listen: __certificate_request_handler
  when: 'xxx' in __certificate_providers
```
Note the use of `listen: __certificate_request_handler` - this is how you have multiple handlers for a single task - the `certificate_request` task will then use `notify: __certificate_request_handler`

Reply:
Added here: 0569e988dba550ec6af56df9c1718422069040ef